fetch.test.js

```diff
@@ -0,0 +1,78 @@
+import test from 'ava';
+import sinon from 'sinon';
+
+import { fetchJSON, getJSON, patchJSON, postJSON, putJSON, deleteJSON } from './fetch';
+
+const DEFAULT_RESULT = {
+    status: 200,
+    headers: { 'content-type': 'application/json' },
+    text: () => '{"id": 42, "email": "ozzy@example.com", "name": "Ozzy"}'
+};
+
+test.beforeEach(t => {
+    // Create a mock global window object with a fake fetch method.
+    // Pitfall: This omits ava's built-in isolated context and prevents us from running tests concurrently.
+    // As a result, tests need to be run serially!
+    global.window = {
+        fetch: sinon.fake.resolves(DEFAULT_RESULT)
+    };
+});
+
+test.afterEach.always(t => {
+    // Remove the mock global window object after each test is run.
+    delete global.window;
+});
+
+test.serial('Call `window.fetch` with given url, HTTP method, credentials, and body)', t => {
```

Comment:
Great tests 👍. Is it possible to run them async or did that not work?

Reply:
Serialized tests don't run concurrently, but you can still test async code – see [this async test](https://github.com/datawrapper/shared/pull/2/commits/7e5d1f3e3bc259f359359ef31ef6920ad97c12e7#diff-f3e7998496d8bdfb193355c6d9357d46R34) for an example.

Reply:
Oh sorry that's not what I meant. Can we run them concurrently with `test` instead of `test.serial`?

Reply:
I've refactored the tests, using `test.serial` is no longer necessary.